### ISSUE TITLE
`apk fetch` fails on some packages with `ERROR: {package_name}: BAD ARCHIVE`

### ISSUE THREAD

**Issue by MthwRobinson** [wolfi-dev/os#19795]
`apk fetch` fails in `cgr.dev/chainguard/wolfi-base` for the following packages with a `BAD ARCHIVE` error, even though `apk add` runs successfully. For example:

```
/home/nonroot/apk-packages2 # apk fetch libgcc
Downloading libgcc-13.2.0-r6
ERROR: libgcc-13.2.0-r6: BAD archive
```

I would expect to be able to run `apk add {package}` and see the `.apk` file downloaded to the working directly. This works successfully with `apk fetch bash`.

The context for this request is that I am working to build an image in an environment that does not have access to the Wolfi repository, and so we'd like to download the `.apk` files so they're available for us to install locally.

```
py3.11-pip-24.0-r1
py3.11-setuptools-69.5.1-r1
python-3.11-base-3.11.9-r2
libuv-1.48.0-r1
mesa-glapi-24.0.7-r0
libpciaccess-0.18.1-r0
libx11-1.8.9-r1
libxau-1.0.11-r5
python-3.12-3.12.3-r2
readline-8.2-r6
freetype-2.13.2-r3
libcom_err-1.47.0-r3
libffi-3.4.6-r1
libstdc++-13.2.0-r6
libgcc-13.2.0-r6
libnghttp2-14-1.62.1-r0
libunistring-1.2-r1
libverto-0.3.2-r2
```

**Comment by ajayk** [comment]
thought we had fixed this  as far as I remember we had a weird apk-tools bug that was causing the mismatch between some of the check sums 

cc: @jonjohnsonjr

**Comment by jonjohnsonjr** [comment]
Yeah see here: https://gitlab.alpinelinux.org/alpine/apk-tools/-/issues/10997

**Comment by MthwRobinson** [comment]
Is there a known version `apk-tools` available that doesn't produce bad archive error? I tried with `2.12.11` but still got the same error. If not we can try to build the latest version since the issue from above says it's fixed on `2.14-stable`.

```
/home/nonroot # apk list apk-tools
apk-tools-2.12.11-r2 aarch64 {apk-tools} (GPL-2.0-only) [installed]
apk-tools-2.12.10-r0 aarch64 {apk-tools} (GPL-2.0-only)
apk-tools-2.12.11-r0 aarch64 {apk-tools} (GPL-2.0-only)
apk-tools-2.12.11-r1 aarch64 {apk-tools} (GPL-2.0-only)
apk-tools-2.14.0-r0 aarch64 {apk-tools} (GPL-2.0-only) [upgradable from: apk-tools-2.12.11-r2]
apk-tools-2.14.0_rc1-r0 aarch64 {apk-tools} (GPL-2.0-only) [upgradable from: apk-tools-2.12.11-r2]
apk-tools-2.14.0-r1 aarch64 {apk-tools} (GPL-2.0-only) [upgradable from: apk-tools-2.12.11-r2]
apk-tools-2.14.1-r0 aarch64 {apk-tools} (GPL-2.0-only) [upgradable from: apk-tools-2.12.11-r2]
apk-tools-2.14.1-r1 aarch64 {apk-tools} (GPL-2.0-only) [upgradable from: apk-tools-2.12.11-r2]
apk-tools-2.14.1-r2 aarch64 {apk-tools} (GPL-2.0-only) [upgradable from: apk-tools-2.12.11-r2]
/home/nonroot # apk fetch py3.11-pip
Downloading py3.11-pip-24.0-r1
ERROR: py3.11-pip-24.0-r1: BAD archive
```

**Comment by MthwRobinson** [comment]
Going to close this issue. We were able to work around this by doing `mkdir -p /var/cache/apk && ln -s /var/cache/apk /etc/apk/cache` to set up `apk cache ` and then `apk cache download <package>` to cache the package `/var/cache/apk`.

Thanks for the support and all the great work you do!
